graylog2-server/src/main/java/org/graylog2/plugin/BaseConfiguration.java

```diff
@@ -129,6 +137,14 @@ public WaitStrategy getInputBufferWaitStrategy() {
         return getWaitStrategy(inputBufferWaitStrategy, "inputbuffer_wait_strategy");
     }
 
+    public Path getBinPath() {
+        return binPath;
+    }
+
+    public Path getDataDir() {
```

Comment:
Shouldn't this be used as a parent directory for `message_journal_dir` and `content_packs_dir` by default?

Reply:
As written in https://github.com/Graylog2/graylog2-server/pull/4656#issuecomment-373128648 I deliberately didn't touch the existing settings. I tried to do that in #4656 but there was lots of room for error for users and developers.

Using the `data_dir` as a parent for e.g. `message_journal_dir` requires either injecting both values in every place and manually building the real path or something like I build in #4656 which turned out to be complicated and brittle.

Having e.g. `message_journal_dir` use `data_dir` as base directory if it's a relative path but not using `data_dir` as base if it's an absolute path is pretty confusing for users. (IMHO) Always using `data_dir` as base path also doesn't make sense because then we could just remove the `message_journal_dir` setting. (as mentioned in the second paragraph in https://github.com/Graylog2/graylog2-server/pull/4656#issuecomment-373109961) But that would probably be too strict because the user would have to move all data if the journal should be on a separate disk partition.

Please let me know if you have a better idea. :smiley:

Reply:
I simply meant using `getDataDir().resolve("journal")` as default value for `message_journal_dir`. No fancy annotations or magic.

Reply:
The `getDataDir()` accessor is not reachable from the `KafkaJournalConfiguration`.

Reply:
That's a problem we can solve, once we decide what the desired outcome is. 😉

Reply:
Can we run code to set sensible defaults or is that too late?

Reply:
@kroepke I don't think so. That's what led me down this annotation magic path in #4656.

Reply:
@joschi True. We have to keep in mind that we also need this for plugins, though.

Reply:
Personally I like setting the default as @joschi suggested, but if it is problematic right now I could live without it for now.

Reply:
We can use the default when we move the `message_journal_dir` setting from the `KafkaJournalConfiguration` into `BaseConfiguration`.